src/test/scala/chiselTests/BundleElementsSpec.scala

```diff
@@ -469,16 +503,26 @@ class BundleElementsSpec extends AnyFreeSpec with Matchers {
 
       val io = IO(new BpipOptionBundle)
       assertElementsMatchExpected(io)(
-        "bpipUIntLazyVal" -> _.bpipUIntLazyVal,
         "bpipOptionUInt" -> _.bpipOptionUInt.get,
         "bpipUIntVar" -> _.bpipUIntVar,
+        "bpipUIntLazyVal" -> _.bpipUIntLazyVal,
         "bpipUIntVal" -> _.bpipUIntVal
       )
     }
 
     ChiselStage.emitCHIRRTL(new ALU(ALUConfig(10, mul = true, b = false)))
   }
 
+  "Bundle elements should be ordered in order of declaration" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new SimpleBundleElemOrder)
+    chirrtl should include("input in1 : { a : UInt<1>, b : UInt<8>}")
+    chirrtl should include("input in2 : { a : UInt<1>, b : UInt<8>}")
+
+    val chirrtl2 = ChiselStage.emitCHIRRTL(new SimpleBundleElemOrderByName)
+    chirrtl2 should include("input in0 : { a : UInt<8>, b : UInt<8>, c : UInt<8>}")
+    chirrtl2 should include("input in1 : { a : UInt<8>, b : UInt<8>, c : UInt<8>}")
+  }
+
```

Comment:
These are good, but it would also be nice to do some  assertElementsMatchExpected as that's the thing that changed (i.e. that you fixed) in this PR.